codespell_lib/tests/test_basic.py

```diff
@@ -488,6 +488,61 @@ def test_ignore_regex_flag(tmpdir, capsys):
     assert cs.main(f.name, r'--ignore-regex=\Wdonn\W') == 1
 
 
+def test_two_word_detect(tmpdir, capsys):
+
+    # test simple two word find
+    dictname = str(tmpdir / 'testdict.txt')
+    with open(dictname, 'w') as f:
+        f.write("will not->won't")
+
+    filename = str(tmpdir / 'flag.txt')
+    with open(filename, 'w') as f:
+        f.write('# I will not be corrected!\n')
+
+    code, stdout, stderr = cs.main(filename, "-D%s" % dictname, std=True)
+    assert "won't" in stdout
+    assert code == 1  # one change found
+
+    # test not finding two word match with punctuation between
+    with open(filename, 'w') as f:
+        f.write('# I will. not be corrected!\n')
+        f.write('# I will, not be corrected!\n')
+        f.write('# I will? not be corrected!\n')
+        f.write('# I will! not be corrected!\n')
+        f.write('# I will- not be corrected!\n')
+        f.write('# I will: not be corrected!\n')
+        f.write('# I will; not be corrected!\n')
+        f.write('# I will> not be corrected!\n')
+        f.write('# I will< not be corrected!\n')
+        f.write('# I will@ not be corrected!\n')
+        f.write('# I will# not be corrected!\n')
+        f.write('# I will$ not be corrected!\n')
+        f.write('# I will% not be corrected!\n')
+        f.write('# I will^ not be corrected!\n')
+        f.write('# I will& not be corrected!\n')
+        f.write('# I will* not be corrected!\n')
+        f.write('# I will( not be corrected!\n')
+        f.write('# I will) not be corrected!\n')
+        f.write('# I will_ not be corrected!\n')
+        f.write('# I will+ not be corrected!\n')
+        f.write('# I will= not be corrected!\n')
+        f.write('# I will/ not be corrected!\n')
+        f.write('# I will[ not be corrected!\n')
+        f.write('# I will] not be corrected!\n')
+
+    code, stdout, stderr = cs.main(filename, "-D%s" % dictname, std=True)
+    assert code == 0  # no changes found
+    assert "won't" not in stdout
+
+    # test that it does not match across lines
+    with open(filename, 'w') as f:
+        f.write('# I will\n not be corrected!\n')
+
+    code, stdout, stderr = cs.main(filename, "-D%s" % dictname, std=True)
+    assert code == 0  # no changes found
+    assert "won't" not in stdout
```

Comment:
I ended up not doing matches across line breaks.  This is possible, but the code to switch between two of them is quite different.  (for matching across linebreaks, the extract_pairs object gets constructed outside of the loop and then a extract method is called on each line.  
The way it is currently it resets the object on each line. 
This is fixable, but I don't want to make too much of a mess here.